src/components/completeExercise/directives/completeExerciseExercise/controllers/completeExerciseBaseZnkExercise.controller.js

```diff
@@ -86,6 +84,14 @@
                 }
             }
 
+            function _getQuestionSubjectId(questionCategorForSubject) {
+                if (isSection || !exerciseResult.subjectId){
+                    return CategoryService.getCategoryLevel1ParentSync(questionCategorForSubject);
+                } else  {
+                    return exerciseResult.subjectId;
+                }
+            }
+
```

Comment:
Fix the parameter name "questionCategorForSubject" => "questionCategoriesForSubject"
Send exercise.subjectId as a parameter to this function => "exerciseSubjectId"

This logic doesn't make sense.
Change it so that if we have "exerciseSubjectId", set this to be the question.subjectId, otherwise call the category service function.  In other words - remove the if isSection